### DISCUSSION TITLE
How to show hide resources based on roles/permission while using advance navigation customization

### DISCUSSION THREAD

**Discussion by valpuia** [filamentphp/filament#6004] [Help] [answered]
For roles and permission I used [Shield](https://github.com/bezhanSalleh/filament-shield) plugin.

And for language switch [Language Switch](https://filamentphp.com/plugins/language-switch) plugin.

If I create the side bar navigation like normal, The position changed based on language change, then I used [Advance Navigation Customization](https://filamentphp.com/docs/2.x/admin/navigation#advanced-navigation-customization). The position changed is resolved means they stay on their position after language changed.

But I got another problem, the sidebar navigation always show if the user cannot access the particular resource also.

How to make the sidebar navigation based on the roles, like normal navigation (in normal navigation, they show/hide based on their roles and permission as expected)

**Comment by danharrin** [comment, marked as answer]
You should check `::shouldRegisterNavigation()` on resources before registering them in the navigation

**Reply by valpuia** [reply]
I got `Method App\Filament\Resources\PostResource::shouldRegisterNavigation does not exist.` can you briefly explain how to use, you can see my code below inside 'AppServiceProvider'

```
use App\Filament\Resources\PostResource;
use Filament\Facades\Filament;
use Filament\Navigation\NavigationBuilder;
use Filament\Navigation\NavigationGroup;

Filament::navigation(function (NavigationBuilder $builder): NavigationBuilder {
    $builder->items([
        ...Dashboard::getNavigationItems(),
    ]);

    $builder->groups([
        NavigationGroup::make(__('message.post.management'))
            ->items([
                // ...PostResource::getNavigationItems(), // previous registration (cannot hide/show base on role-permission)
                ...PostResource::shouldRegisterNavigation(), // throw above error here
            ]),
    ]);

    return $builder;
});
```

**Reply by danharrin** [reply]
Ah I just realised that method is protected in v2. Try this instead:

```php
use App\Filament\Resources\PostResource;
use Filament\Facades\Filament;
use Filament\Navigation\NavigationBuilder;
use Filament\Navigation\NavigationGroup;

Filament::navigation(function (NavigationBuilder $builder): NavigationBuilder {
    $builder->items([
        ...Dashboard::getNavigationItems(),
    ]);

    $builder->groups([
        NavigationGroup::make(__('message.post.management'))
            ->items([
                ...(PostResource::canViewAny() ? PostResource::getNavigationItems() : []),
            ]),
    ]);

    return $builder;
});
```

**Reply by valpuia** [reply]
It works.. thank you
